### PR TITLE
Implement `MsgCount` for Tigris

### LINKED ISSUE #771: Implement `MsgCount` for Tigris
If compatibility tests are not ready yet, add it to the smoke test.

### PR DESCRIPTION
**Pull request by rumyantseva**
# Description

This PR closes #771.

The implementation is similar to how `MsgCount` is implemented for Postgres and how `MsgFind` is implemented for Tigris.

## Readiness checklist

* [x] I added tests for new functionality or bugfixes.
* [x] I ran `task all`, and it passed.
* [x] I added/updated comments for both exported and unexported top-level declarations (functions, types, etc).
* [ ] I checked comments rendering with `task godocs`.
* [x] (for maintainers only) I set Reviewers, Assignee, Labels, Project and project's Sprint fields.
* [x] I marked all done items in this checklist.

### CODE REVIEW

File: internal/handlers/tigris/fetch.go [hunk 1]
```diff
@@ -38,6 +41,14 @@ func (h *Handler) fetch(ctx context.Context, param fetchParam) ([]*types.Documen
 
 	collection, err := db.DescribeCollection(ctx, param.collection)
 	if err != nil {
+		if status.Code(err) == codes.NotFound {
```
Comment (rumyantseva) [inline internal/handlers/tigris/fetch.go:44]: @AlekSi so, I stayed with gRPC for now, at least it doesn't introduce new dependencies. I'll do a small errors research in a separate PR.

File: internal/handlers/tigris/msg_drop.go [hunk 2]
```diff
@@ -46,11 +46,18 @@ func (h *Handler) MsgDrop(ctx context.Context, msg *wire.OpMsg) (*wire.OpMsg, er
...
 
-	if err = h.driver.UseDatabase(db).DropCollection(ctx, collection); err != nil {
-		if status.Code(err) == codes.NotFound {
+	err = h.driver.UseDatabase(db).DropCollection(ctx, collection)
+	switch err := err.(type) {
+	case nil:
+		// do nothing
+	case *driver.Error:
+		//nolint:nosnakecase // Tigris named their const that way
+		if err.Code == api.Code_NOT_FOUND {
```
Comment (AlekSi) [inline internal/handlers/tigris/msg_drop.go:55]: Maybe extract it into `isNotFound` function to have `//nolint` in a single place?

File: go.mod [hunk 3]
```diff
@@ -18,7 +18,7 @@ require (
 	golang.org/x/exp v0.0.0-20220713135740-79cabaa25d75
 	golang.org/x/net v0.0.0-20220708220712-1185a9018129
 	golang.org/x/sys v0.0.0-20220715151400-c0bba94af5f8
-	google.golang.org/grpc v1.48.0
+	google.golang.org/grpc v1.48.0 // indirect
```
[the remark this answers is not in the document]
Comment (rumyantseva) [reply go.mod]: fixed

File: internal/handlers/tigris/msg_drop.go (hunk 2 above)
Comment (rumyantseva) [reply internal/handlers/tigris/msg_drop.go]: fixed

File: Taskfile.yml [hunk 4]
```diff
@@ -129,19 +129,25 @@ tasks:
...
+      - >
+        go test -count=1 {{if ne OS "windows"}}-race{{end}} -shuffle=on -coverpkg=../...
+        -coverprofile=integration-pg.txt . -handler=pg
 
   test-integration-tigris:
     desc: "Run integration tests for Tigris handler"
-    dir: integration/tigris
+    dir: integration
     cmds:
-      - go test -count=1 {{if ne OS "windows"}}-race{{end}} -tags=ferretdb_tigris -shuffle=on -coverprofile=../integration-tigris.txt -coverpkg=../... -handler=tigris
```
Comment (rumyantseva) [inline Taskfile.yml:138]: Oh! That wasn't me :/

File: Taskfile.yml [hunk 5]
```diff
@@ -129,19 +129,25 @@ tasks:
     desc: "Run integration tests for PostgreSQL handler"
     dir: integration
     cmds:
-      - go test -count=1 {{if ne OS "windows"}}-race{{end}} -shuffle=on -coverprofile=integration-pg.txt -coverpkg=../... -handler=pg
+      - >
```
Comment (rumyantseva) [inline Taskfile.yml:132]: Codecov fix by @AlekSi

**Review by w84thesun — DISMISSED** [review]
LGTM

File: internal/handlers/tigris/msg_count.go [hunk 6]
```diff
@@ -16,13 +16,110 @@ package tigris
...
+		"returnKey",
+		"showRecordId",
+		"tailable",
+		"oplogReplay",
+		"noCursorTimeout",
+		"awaitData",
+		"allowPartialResults",
+		"collation",
+		"allowDiskUse",
+		"let",
```
Comment (seeforschauer) [inline internal/handlers/tigris/msg_count.go:46]: could you please clarify where did you take all that parameter list bcz I don't see them in `count` unimplemented fields for Postgres backend and also I don't see it in the issue description and in the `count` description for MongoDB.

File: internal/handlers/tigris/msg_count.go [hunk 7]
```diff
@@ -16,13 +16,110 @@ package tigris
...
+		return nil, err
+	}
+	ignoredFields := []string{
+		"hint",
+		"batchSize",
+		"singleBatch",
+		"maxTimeMS",
+		"readConcern",
+		"max",
+		"min",
```
Comment (seeforschauer) [inline internal/handlers/tigris/msg_count.go:58]: The same question here,  https://github.com/FerretDB/FerretDB/pull/928/files#r928734809 
Why did you place these fields here?

File: internal/handlers/tigris/msg_count.go (hunk 6 above)
Comment (rumyantseva) [reply internal/handlers/tigris/msg_count.go:46]: They are the same as for Tigris' `MsgFind`: https://github.com/FerretDB/FerretDB/blob/main/internal/handlers/tigris/msg_find.go#L35. From the first sight, it looks reasonable to have the same list, but after your comment I'm not sure anymore :)

File: internal/handlers/tigris/msg_count.go (hunk 7 above)
Comment (rumyantseva) [reply internal/handlers/tigris/msg_count.go:58]: Same answer :)
They are the same as for Tigris' `MsgFind`:
https://github.com/FerretDB/FerretDB/blob/main/internal/handlers/tigris/msg_find.go#L51

File: internal/handlers/tigris/msg_count.go (hunk 6 above)
Comment (AlekSi) [reply internal/handlers/tigris/msg_count.go:46]: `count` command has different arguments: https://www.mongodb.com/docs/manual/reference/command/count/

**Review by AlekSi — CHANGES_REQUESTED** [review]
We should update command's arguments: https://github.com/FerretDB/FerretDB/pull/928#discussion_r928800858

**Review by w84thesun — APPROVED** [review]

**Review by AlekSi — APPROVED** [review]

**Pull request merged**